plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller.php

```diff
@@ -1538,4 +1612,66 @@ protected function get_product_data( $product, $context = 'view' ) {
 
 		return $data;
 	}
+
+	/**
+	 * Get the related products.
+	 * Related products is handled by the wc_get_related_products() core function.
+	 * Currently, the data that define the related products are the categories and tags.
+	 * This endpoint accepts and combines the categories and tags parameters.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return object
+	 */
+	public function get_related_items( $request ) {
+		$id         = $request->get_param( 'id' );
+		$categories = $request->get_param( 'related_categories' );
+		$tags       = $request->get_param( 'related_tags' );
+		$combine    = $request->get_param( 'combine' );
+		$attributes = $request->get_param( 'attributes' );
+
+		/*
+		 * If the `categories` params is defined,
+		 * filter the categories used by the
+		 * wc_get_related_products() function.
+		 */
+		if ( ! empty( $categories ) ) {
+			add_filter(
+				'woocommerce_get_related_product_cat_terms',
+				function( $product_categories ) use ( $categories, $combine ) {
+					if ( ! $combine ) {
+						return $categories;
+					}
+
+					return array_unique( array_merge( $product_categories, $categories ) );
```

Comment:
I am not sure if combining them makes sense here, this means it won't work very well if a user removes categories or tags and re-triggers the related items ( as it would always include the already saved items ).
I think we should just return the $categories here ( same for the $tags filter ).

Reply:
I partially agree with you.
By default, the value of $combine is set to `false`. However, I can imagine a scenario where this value might need to be changed:

* I have a product (shows 👞👞 )
* This product has the `show-shop` category
* I'd like to have all related products but also pass a new `sport` category.

It's like, _I want all products related to my current product, but also, I'd like to add other ones by defining additional criteria_.